sources.go

```diff
@@ -102,3 +103,47 @@ func (s pstateSource) Discover() ([]string, error) {
 
 	return features, nil
 }
+
+////////////////////////////////////////////////////////////////////////////////
+// Network Source
+
+// Implements main.FeatureSource.
+type networkSource struct{}
+
+func (s networkSource) Name() string { return "netid" }
+func (s networkSource) Discover() ([]string, error) {
+	features := []string{}
+	var total_num_vfs int = 0
+	// sysfs of the node mounted as /hostsys inside the pod
+	netInterfaces, err := ioutil.ReadDir("/hostsys/class/net") //does not return . and ..
+	if err != nil {
+		return nil, fmt.Errorf("can't obtain the network interfaces: %s", err.Error())
+	}
+	for _, netInterface := range netInterfaces {
+		//forming the file name : /hostsys/class/net/<network interface>/device/sriov_numvfs
+		filename := "/hostsys/class/net/" + netInterface.Name() + "/device/sriov_numvfs"
+		bytes_received, err := ioutil.ReadFile(filename)
+		if err != nil {
+			// interface does not support SRIOV
```

Comment:
I see you are using  stderrLogger in main.go but in the above comment you are suggesting to use fmt.Errorf(). Can you please let me know your preference.